libs/perun/dialogs/src/lib/change-expiration-dialog/change-expiration-dialog.component.ts

```diff
@@ -61,13 +82,23 @@ export class ChangeExpirationDialogComponent implements OnInit {
     this.expirationControl.setValue(formatDate(this.expirationControl.value,'yyyy-MM-dd','en-GB'));
   }
 
-  private changeExpiration() {
+  private changeMembershipExpiration() {
     // @ts-ignore
     this.expirationAttr.value = this.newExpiration === 'never' ? null : this.newExpiration;
-    this.attributesManagerService.setMemberAttribute({member: this.data.id, attribute: this.expirationAttr}).subscribe(() => {
+    this.attributesManagerService.setMemberAttribute({member: this.data.member.id, attribute: this.expirationAttr}).subscribe(() => {
       this.loading = false;
       this.notificator.showSuccess(this.successMessage);
       this.dialogRef.close(true);
     })
   }
+
+  private changeSponsorshipExpiration() {
+    const expiration = this.newExpiration === 'never' ? null : this.newExpiration;
+
+    this.memberManager.updateSponsorshipValidity(this.data.member.id, this.data.sponsor.user.id, expiration).subscribe(() => {
```

Comment:
I just realized that this issue could be here because i tested it against idm-test instance and you told me that there were some needed to do some adjustments in core because of this,. if so just ignore this comment